app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -436,6 +441,17 @@ def client_authorized?
       authorized
     end
 
+    def get_parent_host(headers)
+      hostnames = headers["HTTP_X_FORWARDED_SERVER"]
+      host = hostnames.split(",")[0] if hostnames
+      host || Facter.value(:fqdn) || SETTINGS[:fqdn]
+    end
```

Comment:
Mind writing a test for this function?  I think you can likely call it directly on the `@controller` variable from a controller test file